### PR TITLE
add support for multi url,  gpu and embedding batch strategy in rag_handler,

### PR DESCRIPTION
**Pull request by dusvyat**
add support for parsing list of urls from column for theissue.ai
support gpu for embeddings
add batching strategy for embeddings

## Description

Run the following query:

```sql
create ML_Engine rag from rag
using 
openai_api_key="";
```

```sql
CREATE MODEL rag_handler_gh_test
from files (select * from mindsdb_gh)
predict answer
USING
   engine="rag",
   embeddings_model_name="all-MiniLM-L6-v2",
   llm_type="openai",
   url_column_name='download_url',
   vector_store_folder_name='test_issue_ai';
```

`url_column_name` is the name of the column which contains urls you want to parse and get embeddings from

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 1]
```diff
@@ -382,6 +384,9 @@ def df_to_documents(
             raise ValueError(
                 f"page_content_column {page_content_column} not in dataframe columns"
             )
+        if page_content_column == url_column_name:
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:387]: user should specify url column name in create

File: mindsdb/integrations/handlers/rag_handler/tests/test_rag.py [hunk 2]
```diff
@@ -177,7 +175,63 @@ def test_qa(self, postgres_mock_handler):
...
+        self.wait_predictor("proj", "test_rag_writer_qa")
+
+        result_df = self.run_sql(
+            """
+            SELECT p.answer
+            FROM proj.test_rag_writer_qa as p
+            WHERE question='What is the best treatment for a cold?'
+        """
+        )
+        assert result_df["answer"].iloc[0]
```
Comment (paxcema) [inline mindsdb/integrations/handlers/rag_handler/tests/test_rag.py:234]: Seems like it'd be good to add a new test for `url_column_name`, though not sure if we can replicate the UI file upload flow with the `files` integration.

**Review by paxcema — APPROVED** [review]
Changes make sense to me, approving (tho if the test is feasible to introduce, then would be great to include that).

File: mindsdb/integrations/handlers/rag_handler/tests/test_rag.py (hunk 2 above)
Comment (dusvyat) [reply mindsdb/integrations/handlers/rag_handler/tests/test_rag.py:234]: good shout, i will include.

I can try adding a df with a url column in it. Should do the trick

**Pull request merged**